cv_bridge/src/cv_mat_sensor_msgs_image_type_adapter.cpp

```diff
@@ -177,15 +189,21 @@ void
 ROSCvMatContainer::get_sensor_msgs_msg_image_copy(
   sensor_msgs::msg::Image & sensor_msgs_image) const
 {
-  sensor_msgs_image.height = frame_.rows;
-  sensor_msgs_image.width = frame_.cols;
-  if (encoding_override_.has_value() && !encoding_override_.value().empty())
-  {
-    sensor_msgs_image.encoding = encoding_override_.value();
+  cv::Mat sourceMat;
+  if (holds_cv_type<cv::Mat>()) {
+    sourceMat = cv_mat();
+  } else if (holds_cv_type<cv::cuda::GpuMat>()) {
+    cv_gpu_mat().download(sourceMat);
+  } else {
+    throw std::runtime_error("unsupported cv type");
   }
-  else
-  {
-    switch (frame_.type()) {
+
+  sensor_msgs_image.height = sourceMat.rows;
+  sensor_msgs_image.width = sourceMat.cols;
+  if (encoding_override_.has_value() && !encoding_override_.value().empty()) {
+    sensor_msgs_image.encoding = encoding_override_.value();
```

Comment:
We should still check if the encoding and cv type+size/width/height+step are compatible and valid.

Reply:
Wouldn't that defeat the purpose of overriding, though? I see this override more of a "trust me, I know what I'm doing" thing.